core/modules/util/IterableFormatter.h

```diff
@@ -0,0 +1,138 @@
+// -*- LSST-C++ -*-
+/*
+ * LSST Data Management System
+ * Copyright 2015 AURA/LSST.
+ *
+ * This product includes software developed by the
+ * LSST Project (http://www.lsst.org/).
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the LSST License Statement and
+ * the GNU General Public License along with this program.  If not,
+ * see <http://www.lsstcorp.org/LegalNotices/>.
+ */
+#ifndef LSST_QSERV_UTIL_CONTAINERFORMATTER_H
+#define LSST_QSERV_UTIL_CONTAINERFORMATTER_H
+ /**
+  * @file
+  *
+  * @brief  Misc. lightweight vector manipulation.
+  *
+  * @author Fabrice Jammes, IN2P3/SLAC
+  */
+
+// System headers
+#include <vector>
+#include <iterator>
+#include <sstream>
+#include <string>
+
+// Third-party headers
+
+// LSST headers
+
+// Qserv headers
+
+namespace lsst {
+namespace qserv {
+namespace util {
+
+/**
+ *  Provide generic output operator for iterable data structures
+ *
+ *  Output format is [a, b, c, ...]
+ *  The elements stored in the iterable data structure
+ *  must provide an ouput operator.
+ *  Examples:
+ *  - works fine for std::vector<std::string>,
+ *  - doesn't work for std::multimap.
+ */
+template <typename Iter>
+class IterableFormatter
+{
+public:
+
+    /**
+     *  Create a printable wrapper for an iterable data structure
+     *
+     *  @param x:       an iterable data structure
+     *  @param first:   start printing from x[first]
+     *  @param open:    opening bracket, NULL is undefined behaviour
+     *  @param close:   closing bracket, NULL is undefined behaviour
+     *  @param sep:     separator between elements, NULL is undefined behaviour
+     */
+    explicit IterableFormatter(Iter begin, Iter end,
+                               char const *const open, char const *const close,
+                               char const *const sep) :
+            _begin(begin), _end(end), _open(open), _close(close), _sep(sep) {
+    }
+
+    /**
+     *  Output operator for IterableFormatter<Iterable>
+     *
+     *  @param out
+     *  @param iterableFormatter
+     *  @return an output stream, with no newline at the end
+     */
+    friend std::ostream& operator<<(std::ostream& os,
+                                    const IterableFormatter<Iter>& self) {
+        os << self._open;
+        auto last =  std::prev(self._end);
```

Comment:
Be very careful with that , this becomes illegal if `_begin == _end`. Also it requires iterator type to be bi-directional, which limits the usefulness of this method. I would rewrite the whole thing as:

    auto it = _begin;
    if (it != _end) {
        os << *it;
        ++it;
    }
    for ( ; it != _end; ++it) {
        os << _sep << *it;
    }